src/journal-remote/journal-upload.c

```diff
@@ -226,6 +230,26 @@ int start_upload(Uploader *u,
                         h = l;
                 }
 
+                char **values;
+                const char *name;
+                ORDERED_HASHMAP_FOREACH_KEY(values, name, arg_headers) {
+                        _cleanup_free_ char *joined = strv_join(values, ", ");
+                        if (!joined)
+                                return log_oom();
+
+                        if (header_value_is_valid(joined))
+                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                                       "Invalid header value %s for %s", joined, name);
```

Comment:
The condition is swapped...

I do not think this should be critical.
Please log in warning level, and ignore the failure.
Also, `joined` is spurious, in most cases, it is too long. Hence, please do not include `joined` in the log message.
```c
if (!header_value_is_valid(joined)) {
        log_warning("Concatenated header value for %s is invalid, ignoring.", name);
        continue;
}
```